### PR TITLE
fix triple bug reporting, detected in E2E test

### PR DESCRIPTION
**Pull request by petod108**
fix problem with test reporting and added more logs in order to understand better the process
 - Fixing bug:
 When executing step "deploying on QA"
 And there is one bug in test project-
 Then after execution in Jira exist three same bugs
 and from those one one is linked with the test Jira ticket.
 
 which was produced by keeping same data structure for every type of test

- added also better logging
- and reducing size of function because the same code is repeated for every type of test (Acceptance, Installation, Integration)

### CODE REVIEW

**Comment by metmajer** [comment]
@petod108 @s2oBCN you are making changes to the core of test handling. Please take the time to properly describe what you are intending to do here — and add a proper name for the issue, too.

**Comment by petod108** [comment]
Fixing bug:
When executing step "deploying on QA" 
And there is one bug in test project-
Then after execution in Jira exist three same bugs
and from those one one is linked with the test Jira ticket.

File: src/org/ods/orchestration/TestStage.groovy [hunk 1]
```diff
@@ -24,50 +24,38 @@ class TestStage extends Stage {
...
-                        acceptance: getTestResults(steps, repo, Project.TestType.ACCEPTANCE),
-                        installation: getTestResults(steps, repo, Project.TestType.INSTALLATION),
-                        integration: getTestResults(steps, repo, Project.TestType.INTEGRATION),
-                    ]
-                ]
+                def data = [:]
+                data.tests = globalData.tests.clone()
+                data.tests.each {
+                    it.value = getTestResults(steps, repo, it.key.capitalize())
+                }
```
[the remark this answers is not in the document]
Comment (metmajer) [reply src/org/ods/orchestration/TestStage.groovy:49]: @petod108 code is principally fine, but at least a comment would help.

File: src/org/ods/orchestration/TestStage.groovy [hunk 2]
```diff
@@ -24,50 +24,41 @@ class TestStage extends Stage {
...
-
-        def globalStructure = [
-            testReportFiles: [],
-            testResults: [
-                testsuites: []
-            ],
-        ]
         def globalData = [
             tests: [
-                acceptance: globalStructure,
```
Comment (petod108) [inline src/org/ods/orchestration/TestStage.groovy:38]: This code was reason for bug behavior

**Review by metmajer — DISMISSED** [review]

**Comment by metmajer** [comment]
@petod108 please create a cherry-pick PR into 4.x as well

**Comment by petod108** [comment]
Yes, I will do.
I found there in this PR is problem with NonCPS

**Comment by petod108** [comment]
I create another branch with PR because I did pull master (because of my E2E testing) and afterward revert, which "destroyed" this branch.

**Pull request closed without merging**